src/extension/checkBoxes.js

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import cx from "classnames";
+
+class CheckBoxes extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      curFields: this.props.searchFields
+    };
+  }
+
+  handleSelect(select) {
+    // when we use setState to set state, it wouldn't change state immediately
+    // so, we need to use other parameter to store new fields
+    const queryFields = this.props.searchFields
+      .map((field) => field.field === select.field ? { ...field, hidden: !(field.hidden || false) } : field)
+
+    this.setState({
+      curFields: queryFields
+    })
+
+    this.props.onSetFields(queryFields)
+  }
+
+  toggleAll(checked) {
+    // select all fields
+    let newFields = this.state.curFields.map((field) => (field.hidden || false) ? { ...field, hidden: false } : field);
+
+    // if all fields are shown, unselect all
+    if (checked) {
+      newFields = newFields.map((field) => !(field.hidden || false) ? { ...field, hidden: true } : field)
```

Comment:
Only when all fields are shown, the `toggle all` checkbox will unselect all fields.